app/models/problem.rb

```diff
@@ -174,6 +174,16 @@ def notices_since_reopened
     notices.after(opened_at).count
   end
 
+  def url
+    Rails.application.routes.url_helpers.app_problem_url(
+      app,
+      self,
+      protocol: Errbit::Config.protocol,
+      host:     Errbit::Config.host,
+      port:     Errbit::Config.port
+    )
+  end
```

Comment:
This might be more of a view concern than a model concern. Putting `url` in the model involves looking up `Rails.application.routes.url_helpers` and supplying protocol, host, and port whereas the controller's already well-equipped to do:

```diff
- url: report.problem.url
+ url: app_problem_url(report.problem.app, report.problem)
```